components/kyma-environment-broker/internal/hyperscaler/account_pool.go

```diff
@@ -38,19 +40,114 @@ type Credentials struct {
 
 type AccountPool interface {
 	Credentials(hyperscalerType Type, tenantName string) (Credentials, error)
+	ReleaseSubscription(hyperscalerType Type, tenantName string) error
+	CountSubscriptionUsages(hyperscalerType Type, tenantName string) (int, error)
+	IsSubscriptionAlreadyReleased(hyperscalerType Type, tenantName string) (bool, error)
 }
 
-func NewAccountPool(secretsClient corev1.SecretInterface) AccountPool {
+func NewAccountPool(secretsClient corev1.SecretInterface, shootsClient gardener_apis.ShootInterface) AccountPool {
 	return &secretsAccountPool{
 		secretsClient: secretsClient,
+		shootsClient:  shootsClient,
 	}
 }
 
 type secretsAccountPool struct {
 	secretsClient corev1.SecretInterface
+	shootsClient  gardener_apis.ShootInterface
 	mux           sync.Mutex
 }
 
+func (p *secretsAccountPool) IsSubscriptionAlreadyReleased(hyperscalerType Type, tenantName string) (bool, error) {
+
+	labelSelector := fmt.Sprintf("tenantName=%s,hyperscalerType=%s", tenantName, hyperscalerType)
+
+	secret, err := getK8SSecret(p.secretsClient, labelSelector)
+
+	if err != nil {
+		return false, errors.Wrapf(err, "error while looking for a secret used by the tenant %s and hyperscaler %s", tenantName, hyperscalerType)
+	}
+
+	if secret == nil {
+		return false, errors.Errorf("accountPool failed to find subscription secret used by the tenant %s and hyperscaler %s", tenantName, hyperscalerType)
+	}
+
+	labelSelector = fmt.Sprintf("shared!=true, released=true, tenantName=%s,hyperscalerType=%s", tenantName, hyperscalerType)
```

Comment:
`released` label name is a little misleading as it implies that it is clean. I guess the better name would be `dirty` or `toRelease`.

Reply:
I think "dirty" is a good name.